src/hotspot/share/opto/vectornode.hpp

```diff
@@ -73,6 +73,14 @@ class VectorNode : public TypeNode {
     return type()->ideal_reg();
   }
 
+  virtual uint hash() const {
+    if (is_commutative_operation()) {
+      return (uintptr_t)in(1) + (uintptr_t)in(2) + Opcode();
```

Comment:
Commutative implies the operation is binary (`req == 3`). Should there be an assert somewhere to ensure it's always the case (ideally, when marking a node as commutative during construction)?

Reply:
Assertion check added.